src/data-workspace/inputs/option-set.js

```diff
@@ -21,21 +37,54 @@ export const OptionSet = ({
     onFocus,
     disabled,
     locked,
+    multi,
 }) => {
+    const { data: metadata } = useMetadata()
+    const optionSet = selectors.getOptionSetById(metadata, optionSetId)
+    // filter out 'null' options
+    const options = optionSet.options.filter((opt) => !!opt)
+    const sortByOptionsOrder = createSortByOptionsOrder(options)
+
+    const parse = (value) => {
+        if (multi) {
+            return (
+                (value &&
+                    value
+                        .sort(sortByOptionsOrder)
+                        .join(MULTI_TEXT_SEPERATOR)) ||
+                ''
+            )
+        } else {
+            // Empty values need an empty string
+            return value || ''
+        }
+    }
```

Comment:
You could consider using an implicit return, conditional chaining and `??` instead of `||`.
```suggestion
    const parse = (value) =>
        multi
            ? value?.sort(sortByOptionsOrder).join(MULTI_TEXT_SEPERATOR) ?? ''
            : value ?? ''
```
I find it a bit more readable.

Reply:
The reason for `value && value.sort` is that `value &&` will not be true in case of empty string. `value?.sort...` will throw on empty string.